ocs_ci/deployment/flexy.py

```diff
@@ -0,0 +1,274 @@
+            if res:
+                break
+        log_fd = open(self.flexy_log_file, "r")
+
+        while True:
+            line = log_fd.readline()
+            if line:
+                logger.info(line.strip())
+            if self.exitThread:
+                break
+        logger.info("Logger thread done printing flexy logs")
+        # exit only this thread
+        sys.exit()
+
+    def build_container_args(self):
+        """
+        Builds most commonly used arguments for flexy container
+
+        """
+        args = list()
+        args.append(f"--log-opt={self.flexy_log_file}")
+        args.append(f"--env-file={self.flex_env_file}")
+        args.append(f"-w={self.flexy_mnt_container_dir}")
+        args.append(
+            f"--mount=type=bind,source={self.flexy_mnt_host_dir},"
+            f"destination=f{self.flexy_mnt_container_dir},relabel=shared"
+        )
+        args.append(
+            f"--mount=type=bind,source={self.flexy_host_private_conf_dir_path},"
+            f"destination={self.flexy_mnt_private_conf_dir},relabel=shared"
+        )
+        args.append(f"{self.flexy_img_url}")
+        return args
+
+    def clone_and_unlock_ocs_private_conf(self):
+        """
+        Clone ocs_private_conf (flexy env and config) repo into
+        flexy_mnt_host_dir
+
+        """
+        clone_repo(self.flexy_private_conf_url, self.flexy_host_private_conf_dir_path)
+        # git-crypt unlock /path/to/keyfile
+        old_cwd = os.getcwd()
+        os.chdir(self.flexy_host_private_conf_dir_path)
+        cp = subprocess.run(
+            f'git-crypt unlock {constants.FLEXY_GIT_CRYPT_KEYFILE}',
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        if cp.returncode:
+            raise exceptions.CommandFailed(cp.stderr)
+        logger.info("Unlocked the git repo")
+        os.chdir(old_cwd)
+
+    def merge_flexy_env(self):
+        """
+        Update the ocs-osp.env file with the user supplied values.
+        This function assumes that flexy-ocs-private repo has been
+        already cloned
+
+        """
+        config_parser = configparser.ConfigParser()
+        # without this,config parser would convert all
+        # uppercase keys to lowercase
+        config_parser.optionxform = str
+        env_file = os.path.join(
+            self.flexy_host_private_conf_dir_path, constants.FLEXY_DEFAULT_ENV_FILE
+        )
+
+        with open(env_file, "r") as fp:
+            # we need to add section to make
+            # config parser happy, this will be removed
+            # while writing back to file
+            file_content = "[root]\n" + fp.read()
+
+        config_parser.read_string(file_content)
+        # Iterate over config_parser keys, if same key is present
+        # in user supplied dict update config_parser
+        for ele in config_parser.items('root'):
+            if ele[0] in config.ENV_DATA:
```

Comment:
Wouldn't be better to provide new section in the configuration, like `FLEXY`, instead of mixing those parameters used to update the flexy env file with other variables in *ENV_DATA* section?

Reply:
ack, I am fine with FLEXY section.